symLogging.py

```diff
@@ -1,29 +1,54 @@
 import sys
 import threading
 import time
+import logging
+import os.path
+from logging.handlers import RotatingFileHandler
 
 gEnableTracing = False
+logger = None
 
 # Whether to pause and launch debugger at CheckDebug breakpoints
 gDebug = False
 
-def SetTracingEnabled(isEnabled):
-  global gEnableTracing
+
+def SetTracingEnabled(isEnabled, logOptions):
+  global gEnableTracing, logger
   gEnableTracing = isEnabled
 
+  filepath = logOptions.get('logPath', '/var/log/snappy')
+  rotatingHandler = RotatingFileHandler(
+              filename=os.path.join(filepath, 'snappy.log'),
+              mode='a',
+              maxBytes=int(logOptions.get("maxFileSize", 1024*1024)),
+              backupCount=int(logOptions.get("maxFiles", 10)))
+
+  streamHandler = logging.StreamHandler()
+
+  fmt = logging.Formatter('%(asctime)s:%(levelname)s:%(message)s')
+  rotatingHandler.setFormatter(fmt)
+  streamHandler.setFormatter(fmt)
+
+  logger = logging.getLogger("Snappy")
+  logger.addHandler(rotatingHandler)
+  logger.addHandler(streamHandler)
+  logger.setLevel(logging.INFO)
+
 def LogTrace(string):
   global gEnableTracing
   if gEnableTracing:
```

Comment:
Yes, it does. But there is the computation done to format the message string just to go into the function call and the logging library suppresses the message output.